va/validation-authority.go

```diff
@@ -71,22 +72,29 @@ func (va ValidationAuthorityImpl) validateSimpleHTTP(identifier core.AcmeIdentif
 			Type:   core.MalformedProblem,
 			Detail: "Identifier type for SimpleHTTP was not DNS",
 		}
+
+		va.log.Debug(fmt.Sprintf("SimpleHTTP [%s] Identifier failure", identifier))
 		return challenge, challenge.Error
 	}
 	hostName := identifier.Value
 
 	// Check for DNSSEC failures for A/AAAA records
 	_, _, err := va.DNSResolver.LookupHost(hostName)
-	if dnssecErr, ok := err.(core.DNSSECError); ok {
-		challenge.Error = &core.ProblemDetails{
-			Type:   core.DNSSECProblem,
-			Detail: dnssecErr.Error(),
-		}
-	} else {
-		challenge.Error = &core.ProblemDetails{
-			Type:   core.ServerInternalProblem,
-			Detail: "Unable to communicate with DNS server",
+	if err != nil {
+		if dnssecErr, ok := err.(core.DNSSECError); ok {
+			challenge.Error = &core.ProblemDetails{
+				Type:   core.DNSSECProblem,
+				Detail: dnssecErr.Error(),
+			}
+		} else {
+			challenge.Error = &core.ProblemDetails{
+				Type:   core.ServerInternalProblem,
+				Detail: "Unable to communicate with DNS server",
+			}
 		}
+		challenge.Status = core.StatusInvalid
+		va.log.Debug(fmt.Sprintf("SimpleHTTP [%s] DNS failure: %s", identifier, err))
```

Comment:
All of these log functions seem like they could be refactorable so that you don't need to type in SimpleHTTP as the module name. Why not write a wrapper that does that?